File: app/java/app/src/main/java/com/example/accounting/infrastructure/in/web/dto/AuditLogResponse.java
package com.example.accounting.infrastructure.in.web.dto;

import com.example.accounting.domain.model.audit.AuditLog;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Map;

/**
 * 監査ログレスポンス DTO
 */
@Data
@Builder
public class AuditLogResponse {
    private Long id;
    private String entityType;
    private String entityId;
    private String action;
    private String userId;
    private String userName;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime timestamp;

    private Map<String, Object> oldValues;
    private Map<String, Object> newValues;
    private Map<String, Object> changes;
    private String reason;
    private String summary;

    /**
     * ドメインモデルから DTO に変換
     *
     * @param auditLog 監査ログ
     * @return DTO
     */
    public static AuditLogResponse from(AuditLog auditLog) {
        return AuditLogResponse.builder()
                .id(auditLog.getId())
                .entityType(auditLog.getEntityType())
                .entityId(auditLog.getEntityId())
                .action(auditLog.getAction().name())
                .userId(auditLog.getUserId())
                .userName(auditLog.getUserName())
                .timestamp(auditLog.getTimestamp())
                .oldValues(auditLog.getOldValues())
                .newValues(auditLog.getNewValues())
                .changes(auditLog.getChanges())
                .reason(auditLog.getReason())
                .summary(auditLog.getSummary())
                .build();
    }
}

File: app/csharp/AccountingSystem.Api/Infrastructure/Web/Dtos/AuditLogResponse.cs
using AccountingSystem.Infrastructure.Persistence.Dapper.Entities;
using AccountingSystem.Domain.Models.Audit;

namespace AccountingSystem.Infrastructure.Web.Dtos;

/// <summary>
/// 監査ログレスポンス DTO
/// </summary>
public class AuditLogResponse
{
    public long? Id { get; set; }
    public string EntityType { get; set; } = string.Empty;
    public string EntityId { get; set; } = string.Empty;
    public string Action { get; set; } = string.Empty;
    public string ActionDisplayName { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public string UserName { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; }
    public Dictionary<string, object>? OldValues { get; set; }
    public Dictionary<string, object>? NewValues { get; set; }
    public Dictionary<string, object>? Changes { get; set; }
    public string? Reason { get; set; }
    public string? IpAddress { get; set; }
    public string? Summary { get; set; }

    public static AuditLogResponse From(AuditLog auditLog)
    {
        return new AuditLogResponse
        {
            Id = auditLog.Id,
            EntityType = auditLog.EntityType,
            EntityId = auditLog.EntityId,
            Action = auditLog.Action.ToString(),
            ActionDisplayName = auditLog.Action.GetDisplayName(),
            UserId = auditLog.UserId,
            UserName = auditLog.UserName,
            Timestamp = auditLog.Timestamp,
            OldValues = auditLog.OldValues,
            NewValues = auditLog.NewValues,
            Changes = auditLog.Changes,
            Reason = auditLog.Reason,
            IpAddress = auditLog.IpAddress,
            Summary = auditLog.GetSummary()
        };
    }
}
